### PR TITLE
Document event mappers

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

Document event mappers

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [x] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

**Review by plousada — APPROVED** [review]

File: docs/advanced_configuration.md [hunk 1]
```diff
@@ -97,6 +97,67 @@ const spanId = await DdTrace.startSpan('foo', { custom: 42 }, Date.now());
...
+);
+config.logEventMapper = event => event;
+config.errorEventMapper = event => event;
+config.resourceEventMapper = event => event;
+config.actionEventMapper = event => event;
+```
+
+Each mapper is a function with a signature of `(T) -> T?`, where `T` is a concrete RUM event type. This allows changing portions of the event before it is sent, or dropping the event entirely.
+
+For example, to redact sensitive information in a RUM Error's `message`, implement a custom `redacted` function and use it in `errorEventMapper`:
```
Comment (rtrieu) [inline docs/advanced_configuration.md:121]:
```suggestion
For example, to redact sensitive information from a RUM error `message`, implement a custom `redacted` function and use it in `errorEventMapper`:
```

File: docs/advanced_configuration.md [hunk 2]
```diff
@@ -97,6 +97,67 @@ const spanId = await DdTrace.startSpan('foo', { custom: 42 }, Date.now());
 DdTrace.finishSpan(spanId, { custom: 21 }, Date.now());
 ```
 
+## Modify or drop RUM events
+
+To modify attributes of a RUM event before it is sent to Datadog or to drop an event entirely, use the Event Mappers API when configuring the RUM React Native SDK:
```
Comment (rtrieu) [inline docs/advanced_configuration.md:102]:
```suggestion
To modify attributes of a RUM event before it is sent to Datadog, or to drop an event entirely, use the Event Mappers API when configuring the RUM React Native SDK:
```

File: docs/advanced_configuration.md [hunk 3]
```diff
@@ -97,6 +97,67 @@ const spanId = await DdTrace.startSpan('foo', { custom: 42 }, Date.now());
...
+```javascript
+config.errorEventMapper = event => {
+    event.message = redacted(event.message);
+    return event;
+};
+```
+
+Returning `null` from the error, resource, or action mapper drops the event entirely; the event is not sent to Datadog.
+
+Depending on the event’s type, only some specific properties can be modified:
```
Comment (rtrieu) [inline docs/advanced_configuration.md:132]:
```suggestion
Depending on the event type, only some specific properties can be modified:
```

**Review by rtrieu — DISMISSED** [review]
Left some minor feedback. Otherwise, it looks good!

**Review by 0xnm — APPROVED** [review]

**Review by xgouchet — APPROVED** [review]

**Pull request merged**